core/domain/auth_jobs_one_off.py

```diff
@@ -39,6 +39,258 @@
 ID_HASHING_FUNCTION = hash
 
 
+class SyncFirebaseAccountsOneOffJob(jobs.BaseMapReduceOneOffJobManager):
+    """One-off job to sync Firebase accounts with Oppia's user accounts.
+
+    This job does not print aggregated success output, because the reduce phase
+    is sharded by auth_id and could potentially generate tens of 1000s of
+    privacy-sensitive outputs. Instead, success is interpreted as an output
+    without any ERRORs.
+
+    This job is idempotent.
+
+    Pre-conditions:
```

Comment:
Do we check these anywhere? I can't see them in the job... which I guess makes sense, but then we need to be really quite careful about the release/deployment process of this because this information is not displayed on the admin page.

Reply:
Right, they aren't checked by the job. The job instructions will make it clear that these pre-conditions need to hold.